galleries/examples/subplots_axes_and_figures/secondary_axis.py

```diff
@@ -96,48 +95,49 @@ def one_over(x):
 plt.show()
 
 # %%
-# Sometime we want to relate the axes in a transform that is ad-hoc from
-# the data, and is derived empirically.  In that case we can set the
-# forward and inverse transforms functions to be linear interpolations from the
-# one data set to the other.
+# Sometime we want to relate the axes in a transform that is ad-hoc from the data, and
+# is derived empirically. Or, one axis could be a complicated nonlinear function of the
+# other. In these cases we can set the forward and inverse transform functions to be
+# linear interpolations from the one set of independent variables to the other.
 #
 # .. note::
 #
 #   In order to properly handle the data margins, the mapping functions
 #   (``forward`` and ``inverse`` in this example) need to be defined beyond the
-#   nominal plot limits.
-#
-#   In the specific case of the numpy linear interpolation, `numpy.interp`,
-#   this condition can be arbitrarily enforced by providing optional keyword
-#   arguments *left*, *right* such that values outside the data range are
-#   mapped well outside the plot limits.
+#   nominal plot limits. This condition can be enforced by extending the
+#   interpolation beyond the plotted values, both to the left and the right,
+#   see ``x1n`` and ``x2n`` below.
 
 fig, ax = plt.subplots(layout='constrained')
-xdata = np.arange(1, 11, 0.4)
-ydata = np.random.randn(len(xdata))
-ax.plot(xdata, ydata, label='Plotted data')
-
-xold = np.arange(0, 11, 0.2)
-# fake data set relating x coordinate to another data-derived coordinate.
-# xnew must be monotonic, so we sort...
-xnew = np.sort(10 * np.exp(-xold / 4) + np.random.randn(len(xold)) / 3)
-
-ax.plot(xold[3:], xnew[3:], label='Transform data')
-ax.set_xlabel('X [m]')
+x1_vals = np.arange(2, 11, 0.4)
+# second independent variable is a nonlinear function of the other.
+# this simple example can be more easily handled without interpolation, and is done
+# this way only for pedagogical purposes.
```

Comment:
I don't think this comment helps with understanding. Also the lack of determinism is why the previous example had the random numbers in it. 

```suggestion
```

Reply:
Agreed!